Source/NSConcreteHashTable.m

```diff
@@ -84,13 +86,13 @@ @interface	NSConcreteHashTable : NSHashTable
  : pointerFunctionsEqual(&M->cb.pf, X.ptr, Y.ptr))
 #define GSI_MAP_RELEASE_KEY(M, X)\
  (M->legacy ? M->cb.old.release(M, X.ptr) \
- : pointerFunctionsRelinquish(&M->cb.pf, &X.ptr))
+  : IS_WEAK(M) ? nil : pointerFunctionsRelinquish(&M->cb.pf, &X.ptr))
 #define GSI_MAP_RETAIN_KEY(M, X)\
  (M->legacy ? M->cb.old.retain(M, X.ptr) \
- : pointerFunctionsAcquire(&M->cb.pf, &X.ptr, X.ptr))
+  : IS_WEAK(M) ? nil : pointerFunctionsAcquire(&M->cb.pf, &X.ptr, X.ptr))
```

Comment:
Could you please provide a short explanation why nil is the correct value here?

Reply:
It isn't particularly correct, but the value is unused.  The correct solution is to rewrite GSIMap to use ARC-like semantics.

Reply:
In particular, I copied the use of `nil` here from other macros with the same structure using GSIMap.